test/zimwriterfs-zimcreatorfs.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * is provided AS IS, WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, and
+ * NON-INFRINGEMENT.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#include <unistd.h>
+#include <iostream>
+#include <magic.h>
+
+#include <zim/file.h>
+#include <zim/fileiterator.h>
+
+#include "gtest/gtest.h"
+
+#include "../src/zimwriterfs/zimcreatorfs.h"
+#include "../src/zimwriterfs/article.h"
+#include "../src/tools.h"
+
+
+// stub from zimwriterfs.cpp
+std::string directoryPath;
+bool inflateHtmlFlag = false;
+bool isVerbose() { return false; }
+magic_t magic;
+
+
+// FIXME: does Google Test provide an entry for initializations on program start?
+static bool libmagic_initialized = false;
```

Comment:
Not exactly. Fixture object is created and destroyed before every test. But libmagic has to be initialized only once.

Reply:
libmagic initialization rewritten to be more C++ way (with class).
Same libmagic initialization code could be a part of test fixture class,  but it will complicate the test without a need- we don't need the same data for several tests (what fixture serves to).